test/elixir/test/search_partition_test.exs

```diff
@@ -0,0 +1,187 @@
+defmodule SearchPartitionTest do
+  use CouchTestCase
+
+  @moduletag :search
+
+  @moduledoc """
+  Test Partition functionality with search
+  """
+
+  def create_search_docs(db_name, pk1 \\ "foo", pk2 \\ "bar") do
+    docs = for i <- 1..10 do
+      id = if rem(i, 2) == 0 do
+        "#{pk1}:#{i}"
+      else
+        "#{pk2}:#{i}"
+      end
+      %{
+        :_id => id,
+        :value => i,
+        :some => "field"
+      }
+    end
+
+    resp = Couch.post("/#{db_name}/_bulk_docs", body: %{:docs => docs} )
+    assert resp.status_code == 201
+  end
+
+  def create_ddoc(db_name, opts \\ %{}) do
+    indexFn = "function(doc) {\n  if (doc.some) {\n    index('some', doc.some);\n }\n}"
+    default_ddoc = %{
+      indexes: %{
+        books: %{
+          analyzer: %{name: "standard"},
+          index: indexFn
+        }
+      }
+    }
+
+    ddoc = Enum.into(opts, default_ddoc)
+
+    resp = Couch.put("/#{db_name}/_design/library", body: ddoc)
+    assert resp.status_code == 201
+    assert Map.has_key?(resp.body, "ok") == true
+  end
+
+  def get_ids (resp) do
+    %{:body => %{"rows" => rows}} = resp
+    Enum.map(rows, fn row -> row["id"] end)
+  end
+
+  @tag :with_partitioned_db
+  test "Simple query returns partitioned search results", context do
+    db_name = context[:db_name]
+    create_search_docs(db_name)
+    create_ddoc(db_name)
+
+    url = "/#{db_name}/_partition/foo/_design/library/_search/books"
```

Comment:
I think that the document id for design document should be in the format of `_design/id` instead of `_design/partition:id`. I used the following bits and looks that test can work
```
jiangphs-MacBook-Pro-2:dreyfus jiangph$ git log --oneline
082d5aa Add partition search limit tests
c3f928c Implement separate limits for partitioned queries
5fe2685 Add partition search tests
6d60d1b Support partitioned queries
96877dc Remove deprecated clauses
1bdca62 Merge pull request #39 from cloudant-labs/fix-function-clause-load-docs
```

Reply:
Use commit `8efe72e`, I got the passed test cases.

```
jiangphs-MacBook-Pro-2:elixir jiangph$ mix test --trace ./test/search_partition_test.exs
Excluding tags: [pending: true]


SearchPartitionTest
  * test Works with bookmarks and limit (1621.9ms)
  * test Cannot do partition query with global search ddoc (1076.7ms)
  * test Only returns docs in partition not those in shard (1572.1ms)
  * test normal search on non-partitioned dbs with over limit (1021.3ms)
  * test normal search on non-partitioned dbs still work (1837.9ms)
  * test normal search on non-partitioned dbs without limit (1197.1ms)
  * test normal search on non-partitioned dbs with limit (1585.7ms)
  * test Cannot do global query with partition view (862.0ms)
  * test Simple query returns partitioned search results (1705.8ms)

PartitionHelpers


Finished in 13.6 seconds
9 tests, 0 failures

Randomized with seed 403854
```